hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/PipelineReportHandler.java

```diff
@@ -90,8 +91,12 @@ public void onMessage(PipelineReportFromDatanode pipelineReportFromDatanode,
       try {
         processPipelineReport(report, dn, publisher);
       } catch (IOException e) {
-        LOGGER.error("Could not process pipeline report={} from dn={}.",
-            report, dn, e);
+        // Avoid NotLeaderException logging which happens when processing
+        // pipeline report on followers.
+        if (!(e instanceof NotLeaderException)) {
```

Comment:
SonarLint: Replace the usage of the `instanceof` operator by a catch block.

Reply:
@adoroszlai Thanks for the report. Curious how you found this Sonar check issue?

Reply:
Got it. Intellij has a plugin.

Reply:
For me when I have installed it locally, it has not shown about this. But anyway addressed this.

Reply:
Thanks for updating the patch.